components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/prompt/GeckoPromptDelegate.kt

```diff
@@ -56,41 +56,65 @@ typealias AC_FILE_FACING_MODE = PromptRequest.File.FacingMode
 internal class GeckoPromptDelegate(private val geckoEngineSession: GeckoEngineSession) :
     PromptDelegate {
 
-    override fun onLoginStoragePrompt(
+    private fun Autocomplete.LoginEntry.toLogin() = Login(
+        guid = guid,
+        origin = origin,
+        formActionOrigin = formActionOrigin,
+        httpRealm = httpRealm,
+        username = username,
+        password = password
+    )
+
+    private fun Login.toLoginEntry() = Autocomplete.LoginEntry.Builder()
+        .guid(guid)
+        .origin(origin)
+        .formActionOrigin(formActionOrigin)
+        .httpRealm(httpRealm)
+        .username(username)
+        .password(password)
+        .build()
+
+    override fun onLoginSave(
         session: GeckoSession,
-        prompt: PromptDelegate.LoginStoragePrompt
+        prompt: PromptDelegate.AutocompleteRequest<Autocomplete.LoginSaveOption>
     ): GeckoResult<PromptResponse>? {
-        fun LoginStorage.LoginEntry.toLogin() = Login(
-            guid = guid,
-            origin = origin,
-            formActionOrigin = formActionOrigin,
-            httpRealm = httpRealm,
-            username = username,
-            password = password
-        )
+        val geckoResult = GeckoResult<PromptResponse>()
+        val onConfirmSave: (Login) -> Unit = { login ->
+            geckoResult.complete(prompt.confirm(Autocomplete.LoginSelectOption(login.toLoginEntry())))
+        }
+        val onDismiss: () -> Unit = {
+            geckoResult.complete(prompt.dismiss())
+        }
 
-        fun Login.toLoginEntry() = LoginStorage.LoginEntry.Builder()
-            .guid(guid)
-            .origin(origin)
-            .formActionOrigin(formActionOrigin)
-            .httpRealm(httpRealm)
-            .username(username)
-            .password(password)
-            .build()
+        geckoEngineSession.notifyObservers {
+            onPromptRequest(
+                PromptRequest.SaveLoginPrompt(
+                    hint = prompt.options[0].hint,
```

Comment:
The pr looks great the only. The only thing that I'm a bit confused is about if we are sure that always we are going to have at least one option

Reply:
Yeah probably not ideal and I can try to make this a bit more safe when I'm adding #7134 but we are already doing this for the logins too, just later https://github.com/mozilla-mobile/android-components/blob/master/components/feature/prompts/src/main/java/mozilla/components/feature/prompts/PromptFeature.kt#L429

Since they are broken up now into save and select now, the SaveLoginPrompt should only ever deal with exactly 1 option

Reply:
👍